src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostDocumentCompletionEndpointTest.cs

```diff
@@ -0,0 +1,487 @@
+             delegatedItemLabels: ["style", "dir"]);
+    }
+
+    [Fact]
+    public async Task TagHelperAttributes_NoAutoInsertQuotes_Completion()
+    {
+        await VerifyCompletionListAsync(
+            input: """
+                This is a Razor document.
+
+                <EditForm $$></EditForm>
+
+                The end.
+                """,
+             completionContext: new RoslynVSInternalCompletionContext()
+             {
+                 InvokeKind = RoslynVSInternalCompletionInvokeKind.Typing,
+                 TriggerCharacter = " ",
+                 TriggerKind = RoslynCompletionTriggerKind.TriggerCharacter
+             },
+             expectedItemLabels: ["FormName", "OnValidSubmit", "@..."],
+             autoInsertAttributeQuotes: false);
+    }
+
+    private async Task VerifyCompletionListAsync(
+        TestCode input,
+        RoslynVSInternalCompletionContext completionContext,
+        string[] expectedItemLabels,
+        string[]? delegatedItemLabels = null,
+        string[]? delegatedItemCommitCharacters = null,
+        string[]? snippetLabels = null,
+        bool autoInsertAttributeQuotes = true,
+        bool commitElementsWithSpace = true)
+    {
+        var document = await CreateProjectAndRazorDocumentAsync(input.Text);
+        var sourceText = await document.GetTextAsync(DisposalToken);
+
+        var clientSettingsManager = new ClientSettingsManager([], null, null);
+        clientSettingsManager.Update(ClientAdvancedSettings.Default with { AutoInsertAttributeQuotes = autoInsertAttributeQuotes, CommitElementsWithSpace = commitElementsWithSpace });
+
+        VSInternalCompletionList? response = null;
+        if (delegatedItemLabels is not null)
+        {
+            response = new VSInternalCompletionList()
+            {
+                Items = delegatedItemLabels.Select((label) => new VSInternalCompletionItem()
+                {
+                    Label = label,
+                    CommitCharacters = delegatedItemCommitCharacters,
+                    // If test specifies not to commit with space, set kind to element since we remove space
+                    // commit from elements only. Otherwise test doesn't care, so set to None
+                    Kind = !commitElementsWithSpace ? CompletionItemKind.Element : CompletionItemKind.None,
+                }).ToArray(),
+                IsIncomplete = true
+            };
+        }
+
+        var requestInvoker = new TestLSPRequestInvoker([(Methods.TextDocumentCompletionName, response)]);
+
+        var snippetCompletionItemProvider = new SnippetCompletionItemProvider(new SnippetCache());
+        if (snippetLabels is not null)
+        {
+            var snippetInfos = snippetLabels.Select(label => new SnippetInfo(label, label, label, string.Empty, SnippetLanguage.Html)).ToImmutableArray();
+            snippetCompletionItemProvider.SnippetCache.Update(SnippetLanguage.Html, snippetInfos);
+        }
+;
```

Comment:
```suggestion

```

Reply:
Removed, looks like an accidental typo. FWIW, yet another thing Format Document doesn't fix and I obviously simply didn't notice. No intention of being sloppy here.